src/actions/kloudio/test_kloudio.ts

```diff
@@ -0,0 +1,44 @@
+import * as chai from "chai"
+import * as Hub from "../../hub"
+
+import { KloudioAction } from "./kloudio"
+
+const action = new KloudioAction()
+
+describe(`${action.constructor.name} unit tests`, () => {
+
+  describe("action", () => {
+
+    it("errors if the input has no attachment", () => {
+      const request = new Hub.ActionRequest()
+      return chai.expect(action.execute(request)).to.eventually
+        .be.rejectedWith("No attached json.")
+    })
+  })
+
+  describe("form", () => {
+
+    it("has form", () => {
+      chai.expect(action.hasForm).equals(true)
+    })
+
+    it("has form with API and Gsheet param", (done) => {
+      const request = new Hub.ActionRequest()
+      request.params = { aws_access_key: "foo", aws_secret_key: "bar", aws_bucket: "buzz" }
+      const form = action.validateAndFetchForm(request)
+      chai.expect(form).to.eventually.deep.equal({
+        fields: [{
+          label: "API Key",
+          name: "apiKey",
+          required: true,
+          type: "string",
+        }, {
+          label: "Google Sheets URL",
+          name: "url",
+          required: true,
+          type: "string",
+        }],
+      }).and.notify(done)
+    })
+  })
+})
```

Comment:
We need tests to at least validate what happens when invalid urls or params or used to see what sort of errors and responses are passed back. I don't see any tests for the execute path

Reply:
I have added a couple of unit test cases to check the cases when invalid api key or urls are entered.

Reply:
Hi @phillipperalez! I just wanted to ask whether I need to add any other test cases or the ones currently present are ok?